src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineRewriter.vb

```diff
@@ -271,15 +271,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 ' We'll create proxies for these variable later:
                 ' TODO: so, we have to check if it is already in or not. See the early impl.
 
-                Debug.Assert(initializers.ContainsKey(local))
-                proxy = CreateByRefLocalCapture(typeMap, local, initializers)
-                nonReusableLocalProxies.Add(local, proxy)
+                If initializers.ContainsKey(local) Then
```

Comment:
why did this move from an assert to an 'if check?

Reply:
@CyrusNajmabadi You get 20 test failures, due to the key already being in the collection.
```
 Microsoft.CodeAnalysis.VisualBasic.UnitTests.AsyncTests.Capture_ExceptionInArrayAccess
System.ArgumentException : An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Microsoft.CodeAnalysis.VisualBasic.StateMachineRewriter`1.CaptureLocalSymbol(TypeSubstitution typeMap, LocalSymbol local, ImmutableDictionary`2 initializers) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Lowering\StateMachineRewriter\StateMachineRewriter.vb:line 317
   at Microsoft.CodeAnalysis.VisualBasic.StateMachineRewriter`1.CreateNonReusableLocalProxies(Result captured, Int32& nextFreeHoistedLocalSlot) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Lowering\StateMachineRewriter\StateMachineRewriter.vb:line 181
   at Microsoft.CodeAnalysis.VisualBasic.StateMachineRewriter`1.Rewrite() in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Lowering\StateMachineRewriter\StateMachineRewriter.vb:line 106
   at Microsoft.CodeAnalysis.VisualBasic.AsyncRewriter.Rewrite(BoundBlock body, MethodSymbol method, Int32 methodOrdinal, VariableSlotAllocator slotAllocatorOpt, TypeCompilationState compilationState, DiagnosticBag diagnostics, AsyncStateMachine& stateMachineType) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Lowering\AsyncRewriter\AsyncRewriter.vb:line 100
   at Microsoft.CodeAnalysis.VisualBasic.Rewriter.RewriteIteratorAndAsync(BoundBlock bodyWithoutLambdas, MethodSymbol method, Int32 methodOrdinal, TypeCompilationState compilationState, DiagnosticBag diagnostics, VariableSlotAllocator slotAllocatorOpt, StateMachineTypeSymbol& stateMachineTypeOpt) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Lowering\Rewriter.vb:line 143
   at Microsoft.CodeAnalysis.VisualBasic.Rewriter.LowerBodyOrInitializer(MethodSymbol method, Int32 methodOrdinal, BoundBlock body, SynthesizedSubmissionFields previousSubmissionFields, TypeCompilationState compilationState, Boolean instrumentForDynamicAnalysis, ImmutableArray`1& dynamicAnalysisSpans, DebugDocumentProvider debugDocumentProvider, DiagnosticBag diagnostics, VariableSlotAllocator& lazyVariableSlotAllocator, ArrayBuilder`1 lambdaDebugInfoBuilder, ArrayBuilder`1 closureDebugInfoBuilder, Int32& delegateRelaxationIdDispenser, StateMachineTypeSymbol& stateMachineTypeOpt, Boolean allowOmissionOfConditionalCalls, Boolean isBodySynthesized) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Lowering\Rewriter.vb:line 106
   at Microsoft.CodeAnalysis.VisualBasic.MethodCompiler.LowerAndEmitMethod(MethodSymbol method, Int32 methodOrdinal, BoundBlock block, Binder binderOpt, TypeCompilationState compilationState, DiagnosticBag diagsForCurrentMethod, ProcessedFieldOrPropertyInitializers processedInitializers, SynthesizedSubmissionFields previousSubmissionFields, MethodSymbol constructorToInject, Int32& delegateRelaxationIdDispenser) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Compilation\MethodCompiler.vb:line 1419
   at Microsoft.CodeAnalysis.VisualBasic.MethodCompiler.CompileMethod(MethodSymbol method, Int32 methodOrdinal, Int32& withEventPropertyIdDispenser, Int32& delegateRelaxationIdDispenser, Predicate`1 filter, TypeCompilationState compilationState, ProcessedFieldOrPropertyInitializers processedInitializers, Binder containingTypeBinder, SynthesizedSubmissionFields previousSubmissionFields, MethodSymbol& referencedConstructor) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Compilation\MethodCompiler.vb:line 1236
   at Microsoft.CodeAnalysis.VisualBasic.MethodCompiler.CompileNamedType(NamedTypeSymbol containingType, Predicate`1 filter) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Compilation\MethodCompiler.vb:line 661
   at Microsoft.CodeAnalysis.VisualBasic.MethodCompiler._Closure$__30-0._Lambda$__0() in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Compilation\MethodCompiler.vb:line 503
   at Roslyn.Utilities.UICultureUtilities.<>c__DisplayClass5_0.<WithCurrentUICulture>b__0() in D:\roslyn-AdamSpeight2008\src\Compilers\Core\Portable\InternalUtilities\UICultureUtilities.cs:line 127
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.CodeAnalysis.VisualBasic.MethodCompiler.WaitForWorkers() in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Compilation\MethodCompiler.vb:line 313
   at Microsoft.CodeAnalysis.VisualBasic.MethodCompiler.CompileMethodBodies(VisualBasicCompilation compilation, PEModuleBuilder moduleBeingBuiltOpt, Boolean emittingPdb, Boolean emitTestCoverageData, Boolean hasDeclarationErrors, Predicate`1 filter, DiagnosticBag diagnostics, CancellationToken cancellationToken) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Compilation\MethodCompiler.vb:line 226
   at Microsoft.CodeAnalysis.VisualBasic.VisualBasicCompilation.CompileMethods(CommonPEModuleBuilder moduleBuilder, Boolean emittingPdb, Boolean emitMetadataOnly, Boolean emitTestCoverageData, DiagnosticBag diagnostics, Predicate`1 filterOpt, CancellationToken cancellationToken) in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Portable\Compilation\VisualBasicCompilation.vb:line 2318
   at Microsoft.CodeAnalysis.Compilation.Emit(Stream peStream, Stream metadataPEStream, Stream pdbStream, Stream xmlDocumentationStream, Stream win32Resources, IEnumerable`1 manifestResources, EmitOptions options, IMethodSymbol debugEntryPoint, Stream sourceLinkStream, IEnumerable`1 embeddedTexts, CompilationTestData testData, CancellationToken cancellationToken) in D:\roslyn-AdamSpeight2008\src\Compilers\Core\Portable\Compilation\Compilation.cs:line 2244
   at Roslyn.Test.Utilities.RuntimeEnvironmentUtilities.EmitCompilationCore(Compilation compilation, IEnumerable`1 manifestResources, DiagnosticBag diagnostics, CompilationTestData testData, EmitOptions emitOptions) in D:\roslyn-AdamSpeight2008\src\Test\Utilities\Portable\Compilation\IRuntimeEnvironment.cs:line 216
   at Roslyn.Test.Utilities.RuntimeEnvironmentUtilities.EmitCompilation(Compilation compilation, IEnumerable`1 manifestResources, List`1 dependencies, DiagnosticBag diagnostics, CompilationTestData testData, EmitOptions emitOptions) in D:\roslyn-AdamSpeight2008\src\Test\Utilities\Portable\Compilation\IRuntimeEnvironment.cs:line 194
   at Roslyn.Test.Utilities.Desktop.DesktopRuntimeEnvironment.Emit(Compilation mainCompilation, IEnumerable`1 manifestResources, EmitOptions emitOptions, Boolean usePdbForDebugging) in D:\roslyn-AdamSpeight2008\src\Test\Utilities\Portable\Platform\Desktop\DesktopRuntimeEnvironment.cs:line 200
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationVerifier.Emit(IRuntimeEnvironment testEnvironment, IEnumerable`1 manifestResources, EmitOptions emitOptions) in D:\roslyn-AdamSpeight2008\src\Test\Utilities\Portable\CompilationVerifier.cs:line 138
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationVerifier.Emit(String expectedOutput, Nullable`1 expectedReturnCode, String[] args, IEnumerable`1 manifestResources, EmitOptions emitOptions, Verification peVerify, SignatureDescription[] expectedSignatures) in D:\roslyn-AdamSpeight2008\src\Test\Utilities\Portable\CompilationVerifier.cs:line 103
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.Emit(Compilation compilation, IEnumerable`1 dependencies, IEnumerable`1 manifestResources, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, Action`1 assemblyValidator, Action`1 symbolValidator, EmitOptions emitOptions, Verification verify) in D:\roslyn-AdamSpeight2008\src\Test\Utilities\Portable\CommonTestBase.cs:line 154
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.CompileAndVerifyCommon(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 assemblyValidator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, EmitOptions emitOptions, Verification verify) in D:\roslyn-AdamSpeight2008\src\Test\Utilities\Portable\CommonTestBase.cs:line 68
   at Microsoft.CodeAnalysis.VisualBasic.UnitTests.BasicTestBase.CompileAndVerify(XElement source, IEnumerable`1 allReferences, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, VisualBasicCompilationOptions options, VisualBasicParseOptions parseOptions, EmitOptions emitOptions, Verification verify) in D:\roslyn-AdamSpeight2008\src\Compilers\Test\Utilities\VisualBasic\BasicTestBase.vb:line 179
   at Microsoft.CodeAnalysis.VisualBasic.UnitTests.BasicTestBase.CompileAndVerify(XElement source, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, MetadataReference[] references, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, VisualBasicCompilationOptions options, VisualBasicParseOptions parseOptions, EmitOptions emitOptions, Verification verify, Boolean useLatestFramework) in D:\roslyn-AdamSpeight2008\src\Compilers\Test\Utilities\VisualBasic\BasicTestBase.vb:line 137
   at Microsoft.CodeAnalysis.VisualBasic.UnitTests.AsyncTests.Capture_ExceptionInArrayAccess() in D:\roslyn-AdamSpeight2008\src\Compilers\VisualBasic\Test\Emit\CodeGen\CodeGenAsyncTests.vb:line 3608
```